src/pk_core.c

```diff
@@ -606,13 +626,36 @@ DEF(coreInput,
   RET(VAR_OBJ(line));
 }
 
+DEF(coreExit,
+  "exit([value:int]) -> null\n"
+  "Sets the VM's fiber to NULL, causing the VM to return from execution "
+  "and thereby terminating the process entirely. The exit code of the "
+  "process is the optional argument [value] given to exit() which must be "
+  "between 0 and 255 inclusive. If no argument is given, the exit code is 0 "
+  "by default.") {
+
+  int argc = ARGC;
+  if (argc > 1) { // exit() or exit(val).
+    RET_ERR(newString(vm, "Invalid argument count."));
+  }
+
+  int64_t value = 0;
+  if (argc == 1) {
+    if (!validateIntegerRange(vm, ARG(1), &value, 0, 255, "Argument 1"))
```

Comment:
On, POSIX systems the value will overflow to the range of 0 to 255. But on windows, it supports 32 bit signed integer range exit codes. This validation is unnecessary, just use `exit((int)value);`. Also this `exit()` here is temporary for now to exit from the REPL. However thanks for the effort you put into the validator.

Reply:
Changes made, and no worries.